subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainInstantiationResult.java

```diff
@@ -18,11 +18,36 @@
 
 import org.gradle.internal.jvm.inspection.JvmInstallationMetadata;
 
+import javax.annotation.Nullable;
 import java.util.Optional;
 
-public interface JavaToolchainInstantiationResult {
+public class JavaToolchainInstantiationResult {
```

Comment:
What is the logic behind that change?
Do we really need that in 8.0 / 7.6.1 as well?
Trying to minimize what we change last minute.

Reply:
It was an interface with a default implementation, but both were being used in the same area without any real benefits for separating them. Both were internal and quite local, so I've merged them.